dandischema/datacite/__init__.py

```diff
@@ -111,15 +111,37 @@ def to_datacite(
     }
     # meta has also attribute url, but it often empty
     attributes["url"] = str(meta.url or "")
-    # assuming that all licenses are from SPDX?
-    attributes["rightsList"] = [
-        {
-            "schemeUri": "https://spdx.org/licenses/",
-            "rightsIdentifierScheme": "SPDX",
-            "rightsIdentifier": el.name,
-        }
-        for el in meta.license
-    ]
+
+    # Construct the values for the `"rightsList"`
+    #   currently only SPDX licenses are supported
+    rights_list = []
+    license_pattern = re.compile(r"^([^:\s]+):(\S+)$")
+    for license_ in meta.license:
+        license_match = license_pattern.match(license_.value)
+        assert (
+            license_match
+        ), 'License is not of the expected format of "scheme:identifier"'
+        scheme, identifier = license_match.groups()
+        assert all(
+            [scheme, identifier]
+        ), "License scheme and identifier must both exist and be non-empty"
+
+        if scheme.upper() == "SPDX":
+            # SPDX license
+            rights_list.append(
+                {
+                    "rightsIdentifier": identifier,
+                    "rightsIdentifierScheme": "SPDX",
+                    "schemeUri": "https://spdx.org/licenses/",
+                }
+            )
+        else:
+            raise NotImplementedError(
+                f"License scheme {scheme} is not supported. "
+                "Currently only SPDX licenses are supported."
+            )
+    attributes["rightsList"] = rights_list
```

Comment:
please move into a helper function like `_licenses_to_rights_list` or alike to be unittested and avoid long spagetti code (we might move more of similar code later on)

Reply:
`_licenses_to_rights_list()` created with tests added.